src/beacon-api/app/services/thingspeak_sync_service.py

```
@@ -0,0 +1,454 @@
+            while True:
+                url = (
+                    f"https://thingspeak.com/channels/{channel_id}/feeds.json?"
+                    f"start={start_datetime_str}&end={current_end}"
+                    f"&api_key={api_key}&results={_TS_PAGE_SIZE}"
+                )
+
+                try:
+                    response = await client.get(url)
+                    response.raise_for_status()
+                    data = response.json()
+                except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)